src/shared/test/shortcut-test.js

```diff
@@ -0,0 +1,156 @@
+import { createElement, render } from 'preact';
+import { act } from 'preact/test-utils';
+
+import { matchShortcut, installShortcut, useShortcut } from '../shortcut';
+
+function createEvent(key, { ctrl, alt, shift, meta } = {}) {
+  return {
+    key,
+    ctrlKey: !!ctrl,
+    altKey: !!alt,
+    shiftKey: !!shift,
+    metaKey: !!meta,
+  };
+}
+
+describe('shared/shortcut', () => {
+  describe('matchShortcut', () => {
+    [
+      { shortcut: 'ctrl+a', event: createEvent('a', { ctrl: true }) },
+      { shortcut: 'meta+a', event: createEvent('a', { meta: true }) },
+      { shortcut: 'shift+a', event: createEvent('a', { shift: true }) },
+      { shortcut: 'alt+a', event: createEvent('a', { alt: true }) },
+      { shortcut: 'a', event: createEvent('a') },
+    ].forEach(({ shortcut, event }) => {
+      it('should match if modifiers match', () => {
+        assert.isTrue(matchShortcut(event, shortcut));
+      });
+    });
+
+    [{ shortcut: 'ctrl+a', event: createEvent('a', { ctrl: false }) }].forEach(
+      ({ shortcut, event }) => {
+        it('should not match if modifiers do not match', () => {
+          assert.isFalse(matchShortcut(event, shortcut));
+        });
+      }
+    );
+
+    [
+      { shortcut: 'a', event: createEvent('a') },
+      { shortcut: 'enter', event: createEvent('Enter') },
+    ].forEach(({ shortcut, event }) => {
+      it('should match if non-modifier key matches', () => {
+        assert.isTrue(matchShortcut(event, shortcut));
+      });
+    });
+
+    [{ shortcut: 'a', event: createEvent('b') }].forEach(
+      ({ shortcut, event }) => {
+        it('should not match if non-modifier key does not match', () => {
+          assert.isFalse(matchShortcut(event, shortcut));
+        });
+      }
+    );
+
+    ['ctrl', 'META'].forEach(shortcut => {
+      it('should throw an error if no non-modifier key is specified', () => {
+        assert.throws(() => {
+          matchShortcut(createEvent('a'), shortcut);
+        });
+      });
+    });
+  });
+
+  describe('installShortcut', () => {
```

Comment:
Might be good to also have a test where it installs it on a custom element other than document.body, unless I missed that?

Reply:
You're right. I've added a test for this.